core/sql/executor/ExHbaseIUD.cpp

```diff
@@ -472,7 +472,15 @@ ExWorkProcRetcode ExHbaseAccessInsertSQTcb::work()
 	  {
 	    rc = applyPred(scanExpr());
 	    if (rc == 1) // expr is true or no expr
-	      step_ = CREATE_MUTATIONS;
+	      {
+		 rc = evalInsDelPreCondExpr();
```

Comment:
I thought of the preConditionExpr as something similar to the scanExpr. Agree that it is on upqueue though. Do not see any benefit from changing except if we are thinking that some steps are avoided? I can try it out, but would rather do that in a separate checkin if we want to go that route.

About VsbbUpsert, right now the preCondition expression is evaluated only for unique index for which we need CheckAndPut. I am assuming this is always in the SQInsertTcb. Merge on native HBase tables is also not supported yet.